packages/patterns-react/src/patterns/TableOfContents/TableOfContents.js

```diff
@@ -0,0 +1,271 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import { settings as ddsSettings } from '@carbon/ibmdotcom-utilities';
+import { settings } from 'carbon-components';
+import classNames from 'classnames';
+import { featureFlag } from '@carbon/ibmdotcom-utilities';
+import { TABLE_OF_CONTENTS } from '../../internal/FeatureFlags';
+import { TableOfContents20 } from '@carbon/icons-react';
+import { Layout } from '@carbon/ibmdotcom-react';
+
+const { stablePrefix } = ddsSettings;
+const { prefix } = settings;
+
+/**
+ * Table of Contents pattern
+ *
+ * @param {object} props props object
+ * @param {string} props.layoutType layout type property ( 1/3 - 1/4 )
+ * @param {object} props.menuItems menu items object which includes title and id
+ * @param {element} props.children children property of table of contents pattern
+ * @param {object} props.marginTop margin top property which adds margin top value to layout
+ * @param {object} props.marginBottom margin bottom property which adds margin top value to layout
```

Comment:
does the value for `marginBottom` ever match the value for `marginTop` in the specs? Can we have one prop instead to set the two?

Reply:
more of a @wonilsuhibm question, should we explicitly set the margin bottom and top? It's more of a convenience of the layout component, but wondering if it's something that authors would need to ever configure for the table of contents pattern itself?

Reply:
Yeah authors may not even need to configure it. I will talk to Ke and propably remove these settings.

Reply:
removed